src/newLaunch/lbp/stage4.ts

```diff
@@ -0,0 +1,209 @@
+import { WhiteListService } from "services/WhiteListService";
+import { autoinject } from "aurelia-framework";
+import { Router } from "aurelia-router";
+import { DateService } from "services/DateService";
+import { BaseStage } from "newLaunch/baseStage";
+import Litepicker from "litepicker";
+import { Utils } from "services/utils";
+import { EventAggregator } from "aurelia-event-aggregator";
+import { NumberService } from "services/NumberService";
+import { DisclaimerService } from "services/DisclaimerService";
+// import { BigNumber } from "ethers";
+import { Address, EthereumService, fromWei } from "services/EthereumService";
+import { ITokenInfo, TokenService } from "services/TokenService";
+import { TokenListService } from "services/TokenListService";
+
+@autoinject
+export class Stage4 extends BaseStage {
+  startDateRef: HTMLElement | HTMLInputElement;
+  endDateRef: HTMLElement | HTMLInputElement;
+  startDate: Date;
+  startTime = "00:00"
+  endDate: Date;
+  endTime = "00:00";
+  dateService = new DateService();
+  startDatePicker: Litepicker;
+  endDatePicker: Litepicker;
+  lastCheckedFundingAddress: string;
+  fundingSymbol: string;
+  fundingIcon: string;
+  whitelist: Set<Address>;
+  loadingWhitelist = false;
+  lastWhitelistUrlValidated: string;
+  tokenList: Array<string>;
+
+  sliderStartWeights: HTMLInputElement;
+  sliderEndWeights: HTMLInputElement;
+
+  constructor(
+    eventAggregator: EventAggregator,
+    private numberService: NumberService,
+    private ethereumService: EthereumService,
+    router: Router,
+    tokenService: TokenService,
+    private tokenListService: TokenListService,
+    private whiteListService: WhiteListService,
+    private disclaimerService: DisclaimerService,
+  ) {
+    super(router, eventAggregator, tokenService);
+    this.eventAggregator.subscribe("lbp.clearState", () => {
+      this.startDate = undefined;
+      this.endDate = undefined;
+      this.startTime = undefined;
+      this.endTime = undefined;
+    });
+  }
+
+  attached(): void {
+    this.startDatePicker = new Litepicker({
+      element: this.startDateRef,
+      minDate: Date.now(),
+    });
+
+    this.startDatePicker.on("selected", (date: { toJSDate(): Date }) => {
+      this.startDate = date.toJSDate();
+    });
+
+    this.endDatePicker = new Litepicker({
+      element: this.endDateRef,
+      minDate: Date.now(),
+    });
+
+    this.endDatePicker.on("selected", (date: { toJSDate(): Date }) => {
+      this.endDate = date.toJSDate();
+    });
+
+    this.sliderStartWeights.oninput = (event: Event) => {
```

Comment:
you should bind events in the HTML to functions in the viewmodel.  Yes there is an exception for the date picker because it is a third-party component.